pkg/integration/tests/filter_by_changes/type_changes.go

```diff
@@ -0,0 +1,62 @@
+package filter_by_changes
+
+import (
+	"github.com/jesseduffield/lazygit/pkg/config"
+	. "github.com/jesseduffield/lazygit/pkg/integration/components"
+)
+
+var TypeChanges = NewIntegrationTest(NewIntegrationTestArgs{
+	Description:  "Filter commits by author using the typed in author",
+	ExtraCmdArgs: []string{},
+	Skip:         false,
+	SetupConfig: func(config *config.AppConfig) {
+	},
+	SetupRepo: func(shell *Shell) {
+		shell.CreateNCommits(4)
```

Comment:
I'd prefer to explicitly add files and associated commits, just to make this more readable. At the moment the reader has to look up the definition of CreateNCommits to see the relationship between the commit messages and their corresponding file changes. The `shell.CreateNCommits` is really a convenience function for when you just need multiple commits, but in this case we specifically want to be filtering based on the contents of those commits.